src/middlewared/middlewared/plugins/activedirectory.py

```diff
@@ -302,6 +303,67 @@ async def common_validate(self, new, old, verrors):
                 "AD domain name is required."
             )
 
+        if new['allow_dns_updates']:
+            smb = await self.middleware.call('smb.config')
+            addresses = await self.middleware.call(
+                'activedirectory.get_ipaddresses', new, smb, ha_mode
+            )
+
+            if not addresses:
+                verrors.add(
+                    'activedirectory_update.allow_dns_updates',
+                    'No server IP addresses passed DNS validation. '
+                    'This may indicate an improperly configured reverse zone. '
+                    'Review middleware log files for details regarding errors encountered.',
+                )
+
+            for a in addresses:
+                addr = ipaddress.ip_address(a)
+                if addr.is_reserved:
+                    verrors.add(
+                        'activedirectory_update.allow_dns_updates',
+                        f'{addr}: automatic DNS update would result in registering a reserved '
+                        'IP address. Users may disable automatic DNS updates and manually '
+                        'configure DNS A and AAAA records as needed for their domain.'
+                    )
+                    break
```

Comment:
Well, the addresses are auto-detected based on server configuration. The goal here is to catch things before we get started in AD join. For DragonFish we can make the DNS configuration explicit rather than implicit.